src/knitlib/communication/communication.py

```diff
@@ -0,0 +1,59 @@
+# This file is based on Ayab
+
+"""Handles the serial communication protocol.
+"""
+
+import time
+import serial
+
+import logging
+
+
+class Communication(object):
+  """Class Handling the serial communication protocol."""
+
+  def __init__(self, serial=None):
+    """Creates an Communication object, with an optional serial-like object."""
+    logging.basicConfig(level=logging.DEBUG)
+    self.__logger = logging.getLogger(__name__)
+    self.__ser = serial
+
+  def __del__(self):
+    """Handles on delete behavior closing serial port object."""
+    self.close_serial()
+
+  def open_serial(self, pPortname=None):
+    """Opens serial port communication with a portName."""
+    if not self.__ser:
+      self.__portname = pPortname
+      try:
+          self.__ser = serial.Serial(self.__portname, 115200)
```

Comment:
Transfer speed should not be hardcoded. Make it an argument. 
It is 115200 for AYAB, but it is not the same for knittic or disk hack.